mat/cholesky.go

```diff
@@ -308,6 +308,31 @@ func (c *Cholesky) InverseTo(s *SymDense) error {
 	return err
 }
 
+// Scale multiplies the original matrix A by a non-negative constant using
+// its Cholesky decomposition, storing the result in-place into the receiver.
+// That is, if the original Cholesky factorization is
+//  U^T * U = A
+// the updated factorization is
+//  U'^T * U' = f A = A'
+// Scale panics if the constant is non-positive, or if the receiver is non-zero
```

Comment:
s/non-positive/negative/

Reply:
It also panics if the constant is zero though.

Reply:
Then change the wording in the first sentence to "positive"? I'd have the wording in this line as "not positive" rather than "non-positive".